File: mmr/src/main/java/org/halo/thallo/mmr/core/service/DataStoreManager.java
package org.halo.thallo.mmr.core.service;

import org.halo.thallo.mmr.core.model.DataStore;

/**
 * 数据存储空间管理服务
 * Created by lihong on 17-10-10.
 */
public interface DataStoreManager {

    /**
     * 创建DataObject的数据对象存储空间
     * @param oid
     */
    DataStore createStore(String oid) throws MMRException;

    /**
     * 清空数据对象存储空间
     * @param oid DataObject oid
     */
    void emptyStore(String oid);

    /**
     * 删除数据对象存储空间
     * @param oid
     */
    void dropStore(String oid);

    /**
     * 获取表存储空间
     * @param oid
     * @return
     */
    DataStore getDataStore(String oid);
}

File: mmr/src/main/kotlin/org/halo/thallo/mmr/core/service/DataStoreManager.kt
package org.halo.thallo.mmr.core.service


/**
 * 数据管理器
 */
interface DataStoreManager {

    /**
     * 获取表存储空间
     * @param id
     * @return
     */
    fun getStore(id: String): DataStore

    fun get(store: DataStore): DataStore

    /**
     * 保存数据存储配置空间
     * @param id
     * @param store
     * @return
     */
    fun save(store: DataStore): DataStore

    fun update(store: DataStore): DataStore

    fun delete(store: DataStore)

    fun query(filter: Filter, pageRequest: PageRequest) : Page<DataStore>

}